client/taskenv/env.go

```diff
@@ -381,24 +454,23 @@ func NewEmptyBuilder() *Builder {
 	}
 }
 
-// Build must be called after all the tasks environment values have been set.
-func (b *Builder) Build() *TaskEnv {
-	nodeAttrs := make(map[string]string)
+// buildEnv returns the environment variables and device environment
+// variables with respect to the task directories passed in the arguments.
+func (b *Builder) buildEnv(allocDir, localDir, secretsDir string,
```

Comment:
there was a lot of interpolation that happened while creating these environment variables, so i did a back-of-the-envelope dependency analysis and broke out the code for creating the env vars and the device env vars into this method.